alpha/config/cache/kRemoteMemCacheConf.php

```diff
@@ -68,29 +71,47 @@ protected function getRelevantMapList($requestedMapName , $hostname)
 
 	protected function mergeMaps($mapNames)
 	{
-		$mergedMaps = array();
 		$cache = $this->getCache();
-		if(!$cache)
+		if (!$cache)
 		{
 			return null;
 		}
+		$content = null;
+		$globalContent = null;
+		/** Note: we are concatenating the text content to a single ini file content since some inheritence sections are in
+		 * different maps and only after merging them we can create the merged ini file and validate it.
+		 * Since there are also global parameters in many merged files we need to merge them get them seperatly before merging the content
+		 * otherwise they will be merged to the previous map last section and will not be in global section anymore.
+		 */
 		foreach ($mapNames as $mapName)
 		{
-			$map = $cache->get(self::CONF_MAP_PREFIX.$mapName);
-			if($map)
+			$map = $cache->get(self::CONF_MAP_PREFIX . $mapName);
+			if ($map)
 			{
-				$map = json_decode($map,true);
-				if($mergedMaps)
+				$mapContnet = json_decode($map, true);
+				if (is_array($mapContnet))
+				{
+					KalturaLog::debug("Retrieved content in array format from RemoteCache for map - $mapName with content: \n" . print_r($mapContnet, true));
+					$mapContnet = iniUtils::arrayToIniString($mapContnet);
+				}
+				//get global section data - PREG_OFFSET_CAPTURE return offset starting point in index[1] of match
+				preg_match('/\[.*\S.*\]/m', $mapContnet, $matches, PREG_OFFSET_CAPTURE);
+				if (!empty($matches) && isset($matches[0][1]))// find the split point between the global part and the other sections
 				{
-					$mergedMaps = kEnvironment::mergeConfigItem($mergedMaps, $map);
+					$globalContent .= "\n" . substr($mapContnet, 0, $matches[0][1]);
+					$content .= "\n" . substr($mapContnet, $matches[0][1]);
 				}
 				else
 				{
-					$mergedMaps = $map;
+					$globalContent .= "\n" . $mapContnet;
 				}
 			}
 		}
-		return $mergedMaps;
+		$tempIniFile = tempnam(sys_get_temp_dir(), 'TMP_CONF_MAP_');
+		file_put_contents($tempIniFile, $globalContent . "\n" . $content);
```

Comment:
Must make sure here the the file_put_content succeeded otherwise need to alert it